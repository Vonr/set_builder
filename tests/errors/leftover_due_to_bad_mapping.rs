use set_builder::set;

fn main() {
    set! { x : x <- [1, 2, 3], [4, 5, 6], x != 2 };
}
