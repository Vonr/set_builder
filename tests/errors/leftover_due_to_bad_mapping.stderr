error: Found leftover input `, x != 2`,
       perhaps one of your mappings were treated as the predicate due to bad syntax.
        - map: `x`
        - set_mappings: ["x <- [1, 2, 3]"],
        - predicate: `[4, 5, 6]`
 --> tests/errors/leftover_due_to_bad_mapping.rs:4:5
  |
4 |     set! { x : x <- [1, 2, 3], [4, 5, 6], x != 2 };
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `set` (in Nightly builds, run with -Z macro-backtrace for more info)
